### PR TITLE
Changed 3D detected node z position

### PR DESCRIPTION
**Pull request by SamueleSandrini**
In the `detect_3d_node` in the `convert_bb_to_3d` function I modified the calculation of the object's z-position.

The z is not calculated as the depth value at the center of the bounding box, but as the mean depth value corresponding to the object's roi (excluding invalid values such as 0, inf, nan from the calculation). 
This avoids discarding the object's detection just because of a hole in the depth, but attempts to retrieve it anyway. 

In addition, I commented out the remapping` ('detections', 'tracking') `for the `yolov8_detect_3d_node` in the `yolov8_3d.launch.py 
 `which causes a discarding of the detections depending on the tracker configuration.

### CODE REVIEW

**Comment by juandpenan** [comment]
@SamueleSandrini  Nice work I already tested and it works just fine

File: yolov8_bringup/launch/yolov8_3d.launch.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def generate_launch_description():
             ("image_raw", input_image_topic),
             ("depth_image", input_depth_topic),
             ("depth_info", input_depth_info_topic),
-            ("detections", "tracking")
+            # ("detections", "tracking")
```
Comment (juandpenan) [inline yolov8_bringup/launch/yolov8_3d.launch.py]:
```suggestion
             ("detections", "tracking")
```

**Review by juandpenan — COMMENTED** [review]
i think its better to have unique id for the same class

File: yolov8_bringup/launch/yolov8_3d.launch.py (hunk 1 above)
Comment (juandpenan) [reply yolov8_bringup/launch/yolov8_3d.launch.py]: @jmguerreroh  I think with this change is good

Comment (SamueleSandrini) [reply yolov8_bringup/launch/yolov8_3d.launch.py]: I re-added that line

**Comment by SamueleSandrini** [comment]
I re-added that line

**Pull request merged**